src/components/NavigationModal.js

```diff
@@ -0,0 +1,83 @@
+const React = require('react');
+const _merge = require('lodash/merge');
+
+const Icon = require('./Icon');
+
+const StyleConstants = require('../constants/Style');
+
+const NavigationModal = React.createClass({
+  propTypes: {
+    componentStyle: React.PropTypes.object,
+    iconSize: React.PropTypes.number,
+    iconStyle: React.PropTypes.object,
+    iconType: React.PropTypes.string,
+    items: React.PropTypes.arrayOf(React.PropTypes.shape({
+      content: React.PropTypes.node,
+      onClick: React.PropTypes.func
+    })).isRequired,
+    modalItemStyle: React.PropTypes.object,
+    modalStyle: React.PropTypes.object
+  },
+
+  getInitialState () {
+    return {
+      showModal: false
+    };
+  },
+
+  _handleToggle () {
+    this.setState({
+      showModal: !this.state.showModal
+    });
+  },
+
+  render () {
+    const styles = this.styles();
+
+    return (
+      <div onClick={this._handleToggle} style={styles.component}>
+        <Icon size={this.props.iconSize || 30} style={styles.icon} type={this.props.iconType || 'caret-down'} />
+        {this.state.showModal ? (
+          <div style={styles.modal}>
+            {this.props.items.map((item, index) => {
+              return (<div key={index} onClick={item.onClick} style={styles.modalItem}>{item.content}</div>);
```

Comment:
I would separate out things onto their own lines rather than keeping it all on a single line.